tests/unit_tests/dft/source/workspace_external_tests.cpp

```diff
@@ -0,0 +1,358 @@
+    }
+
+    sycl::free(deviceFwd, sycl_queue);
+    sycl::free(deviceBwd, sycl_queue);
+    sycl::free(workspace, sycl_queue);
+    return sanityCheckPasses ? !::testing::Test::HasFailure() : ::testing::Test::HasFailure();
+    ;
+}
+
+template <oneapi::mkl::dft::precision prec, oneapi::mkl::dft::domain dom>
+int test_workspace_external_buffer_impl(std::size_t dftSize, sycl::device* dev) {
+    using namespace oneapi::mkl::dft;
+    using scalar_t = std::conditional_t<prec == precision::DOUBLE, double, float>;
+    using forward_t = std::conditional_t<dom == domain::COMPLEX, std::complex<scalar_t>, scalar_t>;
+    using backward_t = std::complex<scalar_t>;
+
+    sycl::queue sycl_queue(*dev);
+    if (prec == precision::DOUBLE && !sycl_queue.get_device().has(sycl::aspect::fp64)) {
+        return test_skipped;
+    }
+    descriptor<prec, dom> desc(static_cast<std::int64_t>(dftSize));
+
+    desc.set_value(config_param::WORKSPACE_PLACEMENT, config_value::WORKSPACE_EXTERNAL);
+    desc.set_value(config_param::PLACEMENT, config_value::NOT_INPLACE);
+    commit_descriptor(desc, sycl_queue);
+    std::int64_t workspace_bytes = -1;
+    desc.get_value(config_param::WORKSPACE_EXTERNAL_BYTES, &workspace_bytes);
+    if (workspace_bytes < 0) {
+        return ::testing::Test::HasFailure();
+    }
+    sycl::buffer<scalar_t> workspace(static_cast<std::size_t>(workspace_bytes) / sizeof(scalar_t));
+    desc.set_workspace(workspace);
+    // Generate data
+    std::vector<forward_t> hostFwd(static_cast<std::size_t>(dftSize));
+    std::size_t bwdSize = dom == domain::COMPLEX ? dftSize : dftSize / 2 + 1; // TODO: Check this!
+    std::vector<backward_t> hostBwd(bwdSize);
+    rand_vector(hostFwd, dftSize);
+    auto hostFwdCpy = hostFwd; // Some backends modify the input data (rocFFT).
+
+    {
+        sycl::buffer<forward_t> bufFwd(hostFwd);
+        sycl::buffer<backward_t> bufBwd(hostBwd);
+        compute_forward<decltype(desc), forward_t, backward_t>(desc, bufFwd, bufBwd);
+    }
+
+    // To see external workspaces, larger sizes of DFT may be needed. Using the reference DFT with larger sizes is slow,
+    // so use Parseval's theorum as a sanity check instead.
+    bool sanityCheckPasses = parseval_check(dftSize, hostFwdCpy.data(), hostBwd.data());
+
+    if (sanityCheckPasses) {
+        auto hostBwdCpy = hostBwd;
+        {
+            sycl::buffer<forward_t> bufFwd(hostFwd);
+            sycl::buffer<backward_t> bufBwd(hostBwd);
+            compute_backward<decltype(desc), backward_t, forward_t>(desc, bufBwd, bufFwd);
+            sycl_queue.wait_and_throw();
+        }
+        forward_t rescale =
+            static_cast<forward_t>(1) / static_cast<forward_t>(static_cast<scalar_t>(dftSize));
+        sanityCheckPasses = parseval_check(dftSize, hostFwd.data(), hostBwdCpy.data(), rescale);
+    }
+
+    return sanityCheckPasses ? !::testing::Test::HasFailure() : ::testing::Test::HasFailure();
+    ;
```

Comment:
```suggestion
```

Reply:
Done in [1da13ee](https://github.com/oneapi-src/oneMKL/pull/411/commits/1da13eef0fbb663fcadc1abbcc15ad930f977589).